jpyinterpreter-docs/src/modules/ROOT/pages/java-bytecode/java-bytecode.adoc

```diff
@@ -0,0 +1,137 @@
+[[javaBytecode]]
+= Java Bytecode
+
+== Java Virtual Machine Basic Properties
+
+The Java Virtual Machine is a stack based virtual machine.
+It has the following properties:
+
+- Parameters and local variables are stored in slots, with parameters being first.
+This also includes the `this` parameter, which is always in the first slot for instance methods.
+Compilers can also introduce synthetic local variables that do not correspond to any variable in the source.
+
+- On exception, the current stack is cleared and the thrown exception is pushed to the stack.
+
+- Each instruction must have a consistent prior stack size.
+This means if an instruction is the target of multiple instructions, each instruction that have it as a target must have the same post stack size.
+For instance, this is invalid:
++
+```
+LDC 1
+ILOAD 1
+if_icmpeq [if-equal]
+ILOAD 2
+[if-equal]
+RETURN
+```
++
+Because RETURN have an inconsistent prior stack size: if the `if_icmpeq` branch is taken, the expected stack size is 0; otherwise, the expected stack size is 1 (because `ILOAD 2` push an `int` to the stack).
+The bytecode below is valid:
++
+```
+LDC 1
+ILOAD 1
+if_icmpeq [if-equal]
+ILOAD 2
+POP
+[if-equal]
+RETURN
+```
++
+since the expected stack size prior to RETURN is 0 for both branches of `if_icmpeq`.
+
+- The stack is typed, and will cause verification errors if you try to call a method with invalid types on the stack.
+For instance, this is invalid:
++
+```
+ILOAD 1
+INVOKESTATIC [MyClass.method(float) -> float]
+```
++
+since you are calling a method that expects a `float` with an `int`.
+This is also invalid:
++
+```
+ALOAD 1 [Object]
+INVOKEVIRTUAL [MyClass.method() -> void]
+```
++
+Since you trying to call a `MyClass` method on `Object` (which may or may not be a `MyClass` instance).
```

Comment:
Generally, we don't point fingers in documentation ("you").
"We" is preferred, in this case "we are".

I suggest grepping the entire guide for the word "you".